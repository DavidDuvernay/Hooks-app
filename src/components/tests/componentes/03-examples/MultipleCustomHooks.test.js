import React from 'react'
import { useFetch } from '../../../hooks/useFetch'
import { useCounter } from '../../../hooks/useCounter'
const { shallow } = require("enzyme")
const { MultipleCustomHooks } = require("../../../03-examples/MultipleCustomHooks")

jest.mock('../../../hooks/useFetch')
jest.mock('../../../hooks/useCounter')

describe('Pruebas es MultipleCustomHooks', () => {

    useCounter.mockReturnValue({
        counter: 10,
        incremente: () => {}
    })

    test('Debe mostrar el componente', () => {

        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        })

        const wrapper = shallow(<MultipleCustomHooks />);

        expect(wrapper).toMatchSnapshot()
    })

    test('Debe de mostrar la informacion', () => {

          useFetch.mockReturnValue({
            data: [{
                author: 'David',
                quote: 'Hola Mundo'
            }],
            loading: false,
            error: null
        })

        const wrapper = shallow(<MultipleCustomHooks />);

        expect(wrapper.find('.alert').exists()).toBe(false)
        expect(wrapper.find('.mb-0').text().trim()).toBe('Hola Mundo')
        expect(wrapper.find('footer').text().trim()).toBe('David')

    })
})