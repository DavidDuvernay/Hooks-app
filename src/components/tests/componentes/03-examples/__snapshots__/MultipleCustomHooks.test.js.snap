// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas es MultipleCustomHooks Debe mostrar el componente 1`] = `
<div>
  <h1>
    Breaking Bad quotes
  </h1>
  <hr />
  <div
    className="alert alert-info"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
  >
    Siguiente quote
  </button>
</div>
`;
